source/server/api_listener_impl.cc

```diff
@@ -0,0 +1,48 @@
+#include "server/api_listener_impl.h"
+
+#include "envoy/api/v2/lds.pb.h"
+#include "envoy/api/v2/listener/listener.pb.h"
+#include "envoy/stats/scope.h"
+
+#include "common/http/conn_manager_impl.h"
+#include "common/network/resolver_impl.h"
+#include "common/protobuf/utility.h"
+
+#include "server/drain_manager_impl.h"
+#include "server/listener_manager_impl.h"
+
+#include "extensions/filters/network/http_connection_manager/config.h"
+
+namespace Envoy {
+namespace Server {
+
+ApiListenerImpl::ApiListenerImpl(const envoy::config::listener::v3::Listener& config,
+                                 ListenerManagerImpl& parent, const std::string& name)
+    : config_(config), parent_(parent), name_(name),
+      address_(Network::Address::resolveProtoAddress(config.address())),
+      global_scope_(parent_.server_.stats().createScope("")),
+      listener_scope_(parent_.server_.stats().createScope(fmt::format("listener.api.{}.", name_))),
+      factory_context_(parent_.server_, config_, *this, *global_scope_, *listener_scope_),
+      read_callbacks_(SyntheticReadCallbacks(*this)) {}
+
+HttpApiListener::HttpApiListener(const envoy::config::listener::v3::Listener& config,
+                                 ListenerManagerImpl& parent, const std::string& name)
+    : ApiListenerImpl(config, parent, name) {
+  auto typed_config = MessageUtil::anyConvert<
+      envoy::extensions::filters::network::http_connection_manager::v3::HttpConnectionManager>(
+      config.api_listener().api_listener());
```

Comment:
I think you need to validate after calling anyConvert, please check this. It might be useful to add an anyConvertAndValidate variant and replace that where relevant.

Reply:
Good catch, I created the function. There are several sites where the function call be updated. I'll take that in a separate PR https://github.com/envoyproxy/envoy/issues/9712